app/data_model/answer_store.py

```diff
@@ -0,0 +1,133 @@
+        """
+        answer_to_add = answer.copy()
+
+        if self.exists(answer_to_add):
+            self.update(answer)
+        else:
+            self.answers.append(answer_to_add)
+
+    def update(self, answer):
+        """
+        Update the value of an answer already in the answer store.
+
+        :param answer: A dict of flattened answer details.
+        :return:
+        """
+        for index, existing in enumerate(self.answers):
+            if self.same(existing, answer):
+                self.answers[index]['value'] = answer['value']
+                break
+
+    def get(self, answer):
+        """
+        Returns the value of an answer.
+
+        :param answer: The ids for the answer ro return
+        :return: The value of the answer found
+        """
+        for existing in self.answers:
+            if self.same(answer, existing):
+                return existing['value']
+
+        return None
+
+    def exists(self, answer):
+        """
+        Checks to see if an answer exists in the answer store.
+
+        :param answer: A dict of flattened answer details.
+        :return: True if the answer is in the store, False if not.
+        """
+        return self.count(answer) > 0
+
+    def count(self, answer):
+        """
+        Count of the number of instances of an answer in the answer store.
+
+        :param answer: A dict of flattened answer details.
+        :return: 0 if the answer doesn't exist, otherwise the number of instances.
+        """
+        return len(self.filter(answer))
+
+    @staticmethod
+    def same(first, second):
+        """
+        Check to see if two answers are the same.
+        Two answers are considered the same if they share the same block, question, answer and instance Id.
+
+        :param first: A dict of flattened answer details
+        :param second: A dict of flattened answer details.
+        :return: True if both answers are the same, otherwise False.
+        """
+        return (first['block_id'] == second['block_id'] and
+                first['question_id'] == second['question_id'] and
+                first['answer_id'] == second['answer_id'] and
+                first['answer_instance'] == second['answer_instance'])
+
+    def filter(self, filter_vars):
+        """
+        Find all answers in the answer store for a given set of filter parameter matches.
+
+        :param filter_vars: The filter parameters to match against.
+        :return: A list of answer instances that match the filter.
+        """
+        filtered = []
+        for answer in self.answers:
+            matches = True
+            for k, v in filter_vars.items():
+                if k in ['block_id', 'question_id', 'answer_id', 'answer_instance']:
+                    matches = matches and answer[k] == v
+            if matches:
+                filtered.append(answer)
+        return filtered
+
+    def clear(self):
+        self.answers.clear()
+
+    def map(self, filter_vars=None):
```

Comment:
Maybe filtered could raise/log a warning on filtered variables it doesn't recognise?